font-types/src/pen.rs

```diff
@@ -70,4 +70,17 @@ impl PenCommand {
             PenCommand::Close => pen.close(),
         }
     }
+
+    /// The directly stated - not implied - end point of the command.
+    ///
+    /// Notably, Close does have an end point but it is not directly stated so it returns None.
+    pub fn end_point(&self) -> Option<(f32, f32)> {
+        match *self {
+            PenCommand::MoveTo { x, y } => Some((x, y)),
+            PenCommand::LineTo { x, y } => Some((x, y)),
+            PenCommand::QuadTo { x, y, .. } => Some((x, y)),
+            PenCommand::CurveTo { x, y, .. } => Some((x, y)),
+            PenCommand::Close => None,
+        }
```

Comment:
If you're binding the same set of values in a sequence of match arms, you can combine them. This could be written a bit more succinctly:
```rust
        use PenCommand::*;
        match *self {
            MoveTo { x, y } | LineTo { x, y } | QuadTo { x, y, .. } | CurveTo { x, y, .. } => {
                Some((x, y))
            }
            Close => None,
        }
```
But the current code is also fine.

Reply:
Fancy!

Reply:
Done, ty